File: plugin/src/main/java/dev/mmrl/Process.java
package dev.mmrl;

import android.webkit.JavascriptInterface;

import com.dergoogler.mmrl.webui.interfaces.WXInterface;
import com.dergoogler.mmrl.webui.interfaces.WXOptions;

import org.jetbrains.annotations.NotNull;

public class Process  extends WXInterface {
    public Process(WXOptions wxOptions) {
        super(wxOptions);
    }

    @NotNull
    @Override
    public String getName() {
        return "process";
    }

    @JavascriptInterface
    public String platform() {
        return getOptions().getPlatform().name();
    }

    @JavascriptInterface
    public boolean isAlive() {
        return getOptions().isProviderAlive();
    }
}

File: plugin/src/main/kotlin/dev/mmrl/Process.kt
package dev.mmrl

import android.webkit.JavascriptInterface
import com.dergoogler.mmrl.webui.interfaces.WXInterface
import com.dergoogler.mmrl.webui.interfaces.WXOptions

class Process(wxOptions: WXOptions) : WXInterface(wxOptions) {
    override var name: String = "process"

    @JavascriptInterface
    fun platform(): String {
        return options.platform.name
    }

    @get:JavascriptInterface
    val isAlive: Boolean
        get() = options.isProviderAlive
}